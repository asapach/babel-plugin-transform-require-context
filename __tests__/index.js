'use strict';

const babel = require('babel-core');
const plugin = require('../');

test(`doesn't transform require.context property`, () => {
  const input = `
    const context = require.context;
  `;
  const {code} = babel.transform(input, {plugins: [plugin]});
  expect(code).toMatchSnapshot();
});

test(`transforms require.context() call`, () => {
  const input = `
    const context = require.context('components', true, /\\.html$/);
  `;
  const {code} = babel.transform(input, {plugins: [plugin]});
  expect(code).toMatchSnapshot();
});

test(`transforms require.context() result`, () => {
  const input = `
    var modules = requireAll(require.context("./spec", true, /^\\.\\/.*\\.js$/));
  `;
  const {code} = babel.transform(input, {plugins: [plugin]});
  expect(code).toMatchSnapshot();
});

test(`doesn't transform local require`, () => {
  const input = `
    function test(require) {
      const context = require.context('foo', false);
    }
  `;
  const {code} = babel.transform(input, {plugins: [plugin]});
  expect(code).toMatchSnapshot();
});