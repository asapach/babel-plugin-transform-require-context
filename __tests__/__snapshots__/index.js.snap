// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`doesn't transform local require 1`] = `
"
function test(require) {
  const context = require.context('foo', false);
}"
`;

exports[`doesn't transform require.context property 1`] = `
"
const context = require.context;"
`;

exports[`transforms require.context() call 1`] = `
"
const context = function () {
    function req() {}

    req.keys = function () {
        return [];
    };

    req.resolve = function () {};

    return req;
}();"
`;

exports[`transforms require.context() result 1`] = `
"
var modules = requireAll(function () {
    function req() {}

    req.keys = function () {
        return [];
    };

    req.resolve = function () {};

    return req;
}());"
`;
